src/gazebo_geotagged_images_plugin.cpp

```diff
@@ -555,11 +566,25 @@ void GeotaggedImagesPlugin::_handle_request_camera_settings(const mavlink_messag
         &msg,
         0,                      // time_boot_ms
         CAMERA_MODE_IMAGE,      // Camera Mode
-        NAN,                    // Zoom level
+        100.0 *(_zoom - 1.0)/ (_maxZoom - 1.0),                    // Zoom level //TODO: Output zoom
         NAN);                   // Focus level
     _send_mavlink_message(&msg, srcaddr);
 }
 
+void GeotaggedImagesPlugin::_handle_camera_zoom(const mavlink_message_t *pMsg, struct sockaddr* srcaddr)
+{
+    mavlink_command_long_t cmd;
+    mavlink_msg_command_long_decode(pMsg, &cmd);
+    _send_cmd_ack(pMsg->sysid, pMsg->compid,
+                  MAV_CMD_SET_CAMERA_ZOOM, MAV_RESULT_ACCEPTED, srcaddr);
+
+    _zoom += 0.1 * cmd.param2;
```

Comment:
Is there a way you can put all this in a one liner?

Reply:
Done!